ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/alter/rebuild/AlterMaterializedViewRebuildAnalyzer.java

```diff
@@ -465,21 +467,37 @@ private RelNode applyIncrementalRebuild(RelNode basePlan, RelMetadataProvider md
   protected ASTNode fixUpAfterCbo(ASTNode originalAst, ASTNode newAst, CalcitePlanner.PreCboCtx cboCtx)
           throws SemanticException {
     ASTNode fixedAST = super.fixUpAfterCbo(originalAst, newAst, cboCtx);
+    if (mvRebuildMode == MaterializationRebuildMode.INSERT_OVERWRITE_REBUILD) {
+      return fixedAST;
+    } else if (mvRebuildMode == MaterializationRebuildMode.JOIN_INSERT_REBUILD) {
+      fixUpASTJoinInsertIncrementalRebuild(fixedAST);
+      return fixedAST;
+    }
+
+    MaterializedViewASTBuilder astBuilder;
+    if (AcidUtils.isFullAcidTable(mvTable.getTTable())) {
```

Comment:
please extract this into the helper method: getMaterializedViewASTBuilder()

Reply:
done